hbase-server/src/main/java/org/apache/hadoop/hbase/io/MetricsIO.java

```diff
@@ -24,10 +24,12 @@
 @InterfaceAudience.Private
 public class MetricsIO {
 
+  private static MetricsIO instance;
```

Comment:
This probably should be volatile. Below in your new method, you should do synchronization if it's null so it only gets instantiated once. Example: https://github.com/apache/hbase/blob/master/hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcClientConfigHelper.java#L87

Reply:
> This probably should be volatile. Below in your new method, you should do synchronization if it's null so it only gets instantiated once. Example: https://github.com/apache/hbase/blob/master/hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcClientConfigHelper.java#L87

Update the logic as per suggestion.